packages/components/inputs/select-utils/src/create-select-styles.ts

```diff
@@ -94,13 +94,17 @@ type TState = {
 
 type TDesignTokenName = keyof typeof designTokens;
 
+const upperFirst = (str: string) => str.charAt(0).toUpperCase() + str.slice(1);
+
 const getHorizontalConstraintValue = (
   value?: TProps['minMenuWidth'] | TProps['maxMenuWidth']
 ) => {
-  return (
-    designTokens[`constraint${value}` as TDesignTokenName] ||
-    designTokens.constraint3
-  );
+  const designTokenSuffix =
+    typeof value === 'string' ? upperFirst(value) : value;
+  const constraintValue =
+    designTokens[`constraint${designTokenSuffix}` as TDesignTokenName] ||
+    designTokens.constraint3;
```

Comment:
I think if the current `value` is `auto`, then `constraintValue` will fall back to `designTokens.constraint3` here. Not sure if this is correct. My point is perhaps we don't need the fallback value.

Reply:
Yes, I also have doubts about this. Maybe @FilPob can help us what should the 'auto' prop do in this case?

Reply:
Thanks for pointing that out.

I had another implementation alternative that I discarded at last minute but I think is better.

Please take a look at the changes here: 54bbd00d7ca3185499dc6547e59f4cdfd1e3ea02

Reply:
Thanks, Carlos. I agree, this is a better solution.

Reply:
Honestly I am also not very sure what that `auto` option is doing :D But maybe it doesnt really make sense to have that option available for min.width and max.width.... I can image if you have just one "width" as a prop that the value gets calculated automatically somehow. But setting both min.width and max.width to auto doesnt make sense does it?

Reply:
Let's try to get this merged so we fix the bug and we can customize the available options as a follow-up 🙏